airbyte-ci/connectors/pipelines/pipelines/models/secrets.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+from __future__ import annotations
+
+import hashlib
+import json
+import os
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Dict, List
+
+from dagger import Client as DaggerClient
+from dagger import Secret as DaggerSecret
+from google.cloud import secretmanager_v1
+from google.oauth2 import service_account  # type: ignore
+
+
+class SecretNotFoundError(Exception):
+    pass
+
+
+class SecretString(str):
+    """The use of this string subtype will prevent accidental prints of secret value to the console."""
+
+    @property
+    def _masked_value(self) -> str:
+        return "<SecretString: hidden>"
+
+    def __repr__(self) -> str:
+        return self._masked_value
+
+
+class SecretStore(ABC):
+    @abstractmethod
+    def _fetch_secret(self, name: str) -> str:
+        raise NotImplementedError("SecretStore subclasses must implement a _fetch_secret method")
+
+    def fetch_secret(self, name: str) -> str:
+        return SecretString(self._fetch_secret(name))
+
+
+class GSMSecretStore(SecretStore):
+    def __init__(self, gcp_credentials: Secret) -> None:
+        service_account_info = json.loads(gcp_credentials.value)
+        credentials = service_account.Credentials.from_service_account_info(service_account_info)
+        self.gsm_client = secretmanager_v1.SecretManagerServiceClient.from_service_account_info(service_account_info)
+        # This assumes the service account can only read a single project: the one it was created on.
+        # If we want to read secrets from multiple project we'd have to create a secret mapping (in an env var?)
+        # Which would map secret store aliases to project ids.
+        self.project_id = credentials.project_id
+
+    def _fetch_secret(self, name: str) -> str:
+        request = secretmanager_v1.ListSecretVersionsRequest(
+            parent=f"projects/{self.project_id}/secrets/{name}",
+        )
+        page_result = self.gsm_client.list_secret_versions(request=request)
+        if not page_result:
+            raise SecretNotFoundError(f"No secret found in GSM for secret {name}")
+        for version in page_result:
+            # 1 means enabled version
+            if version.state == 1:
```

Comment:
Can we use [`State.ENABLED`](https://cloud.google.com/python/docs/reference/secretmanager/latest/google.cloud.secretmanager_v1beta2.types.SecretVersion.State)?

Reply:
I could not figure how to import this enum class from the google library 🫤